src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/FormStore.js

```diff
@@ -117,18 +117,31 @@ export default class FormStore {
         this.resourceStore.set(name, value);
     }
 
+    change(name: string, value: mixed) {
+        this.resourceStore.change(name, value);
+    }
+
     @computed get locale(): ?IObservableValue<string> {
         return this.resourceStore.locale;
     }
 
-    @action changeType(type: string) {
+    setType(type: string) {
+        this.saveType(type);
+        this.set(TYPE, type);
+    }
+
+    changeType(type: string) {
+        this.saveType(type);
+        this.change(TYPE, type);
+    }
+
+    @action saveType(type: string) {
```

Comment:
`saveType` is a bit irritating, because it sounds like it is saved on the server, although it isn't... Can't we come up with a better name? Otherwise I could imagine having the check below for the type in a `hasTypes` function and reuse that in the `setType` and `changeType` function, although I can see that this is duplicating some code :confused:

Reply:
Or what about `setInternalType`?